contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,367 @@
+            if (swapTokens[i] == _addr) {
+                revert("Swap token already added");
+            }
+        }
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+
+        // Revert if feed does not exist
+        IOracle(priceProvider).price(_addr);
+
+        swapTokens.push(_addr);
+
+        emit SwapTokenAdded(_addr);
+    }
+
+    /**
+     * @dev Remove a swap token from the tokens that get liquidated for stablecoins.
+     * @param _addr Address of the token
+     */
+    function removeSwapToken(address _addr) external onlyGovernor {
+        uint256 swapTokenIndex = swapTokens.length;
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            if (swapTokens[i] == _addr) {
+                swapTokenIndex = i;
+                break;
+            }
+        }
+
+        require(swapTokenIndex != swapTokens.length, "Swap token not added");
+
+        // Shift everything after the index element by 1
+        for (uint256 i = swapTokenIndex; i < swapTokens.length - 1; i++) {
+            swapTokens[i] = swapTokens[i + 1];
+        }
+        swapTokens.pop();
+
+        emit SwapTokenRemoved(_addr);
+    }
+
+    /**
+     * @dev Add/update a reward token configuration that holds harvesting config variables
+     * @param _tokenAddress Address of the reward token
+     * @param _allowedSlippageBps uint32 maximum allowed slippage denominated in basis points. Example: 300 == 3% slippage
+     * @param _harvestRewardBps uint32 amount of reward tokens the caller of the function is rewarded. Example: 100 == 1%
+     * @param _uniswapV2CompatibleAddr Address Address of a UniswapV2 compatible contract to perform the exchange from reward
+     *        tokens to stablecoin (currently hard-coded to USDT)
+     * @param _liquidationLimit uint256 Maximum amount of token to be sold per one swap function call. When value is 0 there is no limit.
+     */
+    function addRewardTokenConfig(
+        address _tokenAddress,
+        uint32 _allowedSlippageBps,
+        uint32 _harvestRewardBps,
+        address _uniswapV2CompatibleAddr,
+        uint256 _liquidationLimit
+    ) external onlyGovernor {
+        require(
+            _harvestRewardBps <= 1000,
+            "Harvest reward fee should not be over 10%"
+        );
+        require(
+            _allowedSlippageBps <= 1000,
+            "Allowed slippage should not be over 10%"
+        );
+        require(
+            _uniswapV2CompatibleAddr != address(0),
+            "Uniswap compatible address should be non zero address"
+        );
+
+        RewardTokenConfig memory tokenConfig = RewardTokenConfig({
+            harvestRewardBps: _harvestRewardBps,
+            allowedSlippageBps: _allowedSlippageBps,
+            uniswapV2CompatibleAddr: _uniswapV2CompatibleAddr,
+            liquidationLimit: _liquidationLimit
+        });
+
+        rewardTokenConfigs[_tokenAddress] = tokenConfig;
+
+        // Give Uniswap infinite approval when needed
```

Comment:
We also need to remove approvals to the old uniswap if it's a change.

Reply:
Good point adding that in 👍